src/switchroot/ostree-mount-util.h

```diff
@@ -50,58 +54,33 @@ static inline char *
 read_proc_cmdline (void)
 {
   FILE *f = fopen("/proc/cmdline", "r");
-  char *cmdline = NULL;
-  size_t len;
-
   if (!f)
-    goto out;
-
-  /* Note that /proc/cmdline will not end in a newline, so getline
-   * will fail unelss we provide a length.
-   */
-  if (getline (&cmdline, &len, f) < 0)
-    goto out;
-  /* ... but the length will be the size of the malloc buffer, not
-   * strlen().  Fix that.
-   */
-  len = strlen (cmdline);
+    return NULL;
 
-  if (cmdline[len-1] == '\n')
-    cmdline[len-1] = '\0';
-out:
-  if (f)
-    fclose (f);
-  return cmdline;
+  char buf[COMMAND_LINE_SIZE] = {'\0'};
+  size_t len = fread(&buf, 1, sizeof (buf), f);
+  fclose(f);
+  return strndup(buf, len);
 }
 
 static inline char *
 read_proc_cmdline_ostree (void)
 {
-  char *cmdline = NULL;
-  const char *iter;
+  char *cmdline = read_proc_cmdline ();
   char *ret = NULL;
 
-  cmdline = read_proc_cmdline ();
   if (!cmdline)
     err (EXIT_FAILURE, "failed to read /proc/cmdline");
 
-  iter = cmdline;
-  while (iter != NULL)
+  const char* next = strtok (cmdline, " \n");
```

Comment:
Should be we use `strtok_r` instead to be safe? This code might end up getting used in a context where it matters in the future.

Reply:
i was thinking about it, but this function is only called within `switchroot`'s  `main` functions (which are pretty straightforward and simple), and `pthreads` are not used there. 
is it worth doing this code more complex right now?